### PR TITLE
Remove unused XLogAODropSegmentFile function

### PR DESCRIPTION
**Pull request by jimmyyih**
The XLogAODropSegmentFile function was used for implementing invalid
page logging for AO.  In the past year, there have been significant
changes after removing MMXLOG_REMOVE_FILE XLOG record type that make
this function no longer needed (currently only being used by a unit
test).  The most relevant change is that deletion of AO segment files
only happen during DROP TABLE (vacuum was changed to truncate the
segment file) which makes things easier for AO to reuse the existing
invalid page logging implementation.

References:
https://github.com/greenplum-db/gpdb/commit/175c25e8fb0494933087ff19ef29d7377e021702
https://github.com/greenplum-db/gpdb/commit/8165e1b1627b9b2f8b40c3c55eab9de3d137b70a
https://github.com/greenplum-db/gpdb/commit/8838ac983c66af74d0a1422337cf92f08fbe5f2c

### CODE REVIEW

**Comment by BaiShaoqi** [comment]
`When an AO insert is replayed but the segment file does not exist on the mirror, the mirror will PANIC`

Then this is different from heap table. because heap table won't panic on this condition. Is this really what we want ?  Is there a reason for why heap and AO table different in this scenario ?

I thought `XLogAODropSegmentFile`  was used to avoid panic when an AO insert is replayed but the segment file does not exist on the mirror. But I failed to bring this function back.

File: src/backend/access/transam/xlogutils.c [hunk 1]
```diff
@@ -508,13 +510,6 @@ XLogDropRelation(RelFileNode rnode, ForkNumber forknum)
 	forget_invalid_pages(rnode, forknum, 0);
 }
 
-/* Drop an AO/CO segment file from the invalid_page_tab hash table */
-void
-XLogAODropSegmentFile(RelFileNode rnode, uint32 segmentFileNum)
-{
-	forget_invalid_segment_file(rnode, segmentFileNum);
```
Comment (danielgustafsson) [inline src/backend/access/transam/xlogutils.c]: After removing `XLogAODropSegmentFile()`, `forget_invalid_segment_file()` is also dead code and can be removed.

Comment (jimmyyih) [reply src/backend/access/transam/xlogutils.c]: Right, not sure how that didn't get into the PR.  Good catch, thanks.

**Comment by jimmyyih** [comment]
@BaiShaoqi There shouldn't be a difference.  I think the test case that I noted down last year is not correct anymore.  AO and Heap should behave the same for crash recovery and replays.  The `XLogAODropSegmentFile` function is no longer needed because `XLogDropRelation` function clears out the invalid pages correctly.  It was mostly important for AO VACUUM when segment files were dropped.

The crash recovery AO drop redo flow goes like this:
```
xact_redo_commit_internal
    -> DropRelationFiles
        -> XLogDropRelation // forgets logged invalid pages from AO insert transactions that could not find specific AO segment file
        -> smgrdounlinkall -> mdunlink -> mdunlinkfork -> mdunlink_ao // removes all AO files
```

The tracker story you are referencing was more related to how active mirror does consistency checking during WAL replays (the story was very old and not updated).  It's very similar to the crash recovery scenario but revolves around `reachedConsistency` and `minRecoveryPoint` variables.  If the mirror has `reachedConsistency = true` but the mirror tries to log an invalid page, the mirror should PANIC (you can reference the comment at the beginning of `log_invalid_page` function).  This prevents the mirror from holding the logged invalid pages until the end of recovery (e.g. mirror promotion) and PANICing very late.

**Comment by jimmyyih** [comment]
@BaiShaoqi Actually, Heap does behave a little bit differently but not too much.  Replaying heap inserts will go into buffer cache and wait until Checkpointer process to flush to disk.  Replaying AO will immediately go to disk.  For both cases in active mirror consistency checking, it seems like we're okay.

**Comment by BaiShaoqi** [comment]
@jimmyyih 
Thanks for giving so much in detail.

**Pull request merged**